assertj-error-prone/src/main/java/com/palantir/assertj/errorprone/AssertjRefactoring.java

```diff
@@ -0,0 +1,572 @@
+
+    @Override
+    public Description matchSwitch(SwitchTree tree, VisitorState state) {
+        for (AssertjChecker checker : checks) {
+            describe(checker.matchSwitch(tree, state), tree, state);
+        }
+        return Description.NO_MATCH;
+    }
+
+    @Override
+    public Description matchSynchronized(SynchronizedTree tree, VisitorState state) {
+        for (AssertjChecker checker : checks) {
+            describe(checker.matchSynchronized(tree, state), tree, state);
+        }
+        return Description.NO_MATCH;
+    }
+
+    @Override
+    public Description matchThrow(ThrowTree tree, VisitorState state) {
+        for (AssertjChecker checker : checks) {
+            describe(checker.matchThrow(tree, state), tree, state);
+        }
+        return Description.NO_MATCH;
+    }
+
+    @Override
+    public Description matchTry(TryTree tree, VisitorState state) {
+        for (AssertjChecker checker : checks) {
+            describe(checker.matchTry(tree, state), tree, state);
+        }
+        return Description.NO_MATCH;
+    }
+
+    @Override
+    public Description matchTypeCast(TypeCastTree tree, VisitorState state) {
+        for (AssertjChecker checker : checks) {
+            describe(checker.matchTypeCast(tree, state), tree, state);
+        }
+        return Description.NO_MATCH;
+    }
+
+    @Override
+    public Description matchTypeParameter(TypeParameterTree tree, VisitorState state) {
+        for (AssertjChecker checker : checks) {
+            describe(checker.matchTypeParameter(tree, state), tree, state);
+        }
+        return Description.NO_MATCH;
+    }
+
+    @Override
+    public Description matchUnary(UnaryTree tree, VisitorState state) {
+        for (AssertjChecker checker : checks) {
+            describe(checker.matchUnary(tree, state), tree, state);
+        }
+        return Description.NO_MATCH;
+    }
+
+    @Override
+    public Description matchUnionType(UnionTypeTree tree, VisitorState state) {
+        for (AssertjChecker checker : checks) {
+            describe(checker.matchUnionType(tree, state), tree, state);
+        }
+        return Description.NO_MATCH;
+    }
+
+    @Override
+    public Description matchVariable(VariableTree tree, VisitorState state) {
+        for (AssertjChecker checker : checks) {
+            describe(checker.matchVariable(tree, state), tree, state);
+        }
+        return Description.NO_MATCH;
+    }
+
+    @Override
+    public Description matchWhileLoop(WhileLoopTree tree, VisitorState state) {
+        for (AssertjChecker checker : checks) {
+            describe(checker.matchWhileLoop(tree, state), tree, state);
+        }
+        return Description.NO_MATCH;
+    }
+
+    @Override
+    public Description matchWildcard(WildcardTree tree, VisitorState state) {
+        for (AssertjChecker checker : checks) {
+            describe(checker.matchWildcard(tree, state), tree, state);
+        }
+        return Description.NO_MATCH;
+    }
```

Comment:
Given that we expect all of these methods to be the same - would it almost be nicer to do this with a java proxy?

Reply:
Yes and no -- I used reflection to `println` the implementations, but I'm not a huge fan of deferring the cost and complexity of reflection to runtime. We can't simply delegate `Method` invocations because the methods are different (despite having the same name and argument types), and method lookup is relatively expensive.